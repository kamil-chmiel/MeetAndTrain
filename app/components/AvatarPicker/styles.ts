import styled, {css} from 'styled-components/native';

export const AvatarContainer = styled.TouchableOpacity`
  width: 200;
  height: 200;
  margin-top: 30;
  background-color: gray;
  border-radius: 10;
  margin-left: auto;
  margin-right: auto;
`;

export const Avatar = styled.Image`
  flex: 1;
  border-radius: 10;
`;
