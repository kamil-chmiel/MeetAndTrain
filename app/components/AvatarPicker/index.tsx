import React, { Component } from 'react';
import { AvatarContainer, Avatar, PlaceholderText } from './styles';
import ImagePicker from 'react-native-image-picker';

class AvatarPicker extends Component {
	state = {
		avatarSource: null
	};

	onAvatarPress = () => {
		const options = {
			title: 'Select Avatar',
			storageOptions: {
				skipBackup: true,
				path: 'images'
			}
		};

		ImagePicker.showImagePicker(options, (response) => {
			if (response.didCancel) {
				console.log('User cancelled image picker');
			} else if (response.error) {
				console.log('ImagePicker Error: ', response.error);
			} else {
				const source = { uri: response.uri };
				this.setState({
					avatarSource: source
				});
			}
		});
	};

	render() {
		return (
			<AvatarContainer onPress={this.onAvatarPress}>
				<PlaceholderText>Choose your avatar</PlaceholderText>
				<Avatar source={this.state.avatarSource} />
			</AvatarContainer>
		);
	}
}

export default AvatarPicker;
