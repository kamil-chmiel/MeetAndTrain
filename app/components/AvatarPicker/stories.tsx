import React from 'react';
import { View } from 'react-native'
import { storiesOf } from '@storybook/react-native';

import AvatarPicker from '.';

storiesOf('Components/AvatarPicker', module)
.addDecorator(story => <View style={{ flex: 1, justifyContent: 'center' }}>{story()}</View>)
.add('initial', () => <AvatarPicker />)
