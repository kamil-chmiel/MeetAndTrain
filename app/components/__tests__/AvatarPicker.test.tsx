import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import AvatarPicker from '../AvatarPicker'

it('should render correctly', () => {
    const tree = renderer.create(<AvatarPicker />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('should render correctly', () => {
    const tree = renderer.create(<AvatarPicker />).toJSON()
    expect(tree).toMatchSnapshot()
})