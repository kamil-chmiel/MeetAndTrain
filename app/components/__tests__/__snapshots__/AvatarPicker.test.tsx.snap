// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "gray",
      "borderRadius": 10,
      "height": 200,
      "justifyContent": "center",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 30,
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 16,
          "position": "absolute",
        },
      ]
    }
  >
    Choose your avatar
  </Text>
  <Image
    source={null}
    style={
      Array [
        Object {
          "borderRadius": 10,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`should render correctly 2`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "gray",
      "borderRadius": 10,
      "height": 200,
      "justifyContent": "center",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 30,
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 16,
          "position": "absolute",
        },
      ]
    }
  >
    Choose your avatar
  </Text>
  <Image
    source={null}
    style={
      Array [
        Object {
          "borderRadius": 10,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
